service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageHearingsService.java

```diff
@@ -0,0 +1,198 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.document.domain.Document;
+import uk.gov.hmcts.reform.fpl.exceptions.NoHearingBookingException;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+import uk.gov.hmcts.reform.fpl.model.HearingVenue;
+import uk.gov.hmcts.reform.fpl.model.PreviousHearingVenue;
+import uk.gov.hmcts.reform.fpl.model.common.DocmosisDocument;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.docmosis.DocmosisNoticeOfHearing;
+import uk.gov.hmcts.reform.fpl.service.docmosis.DocmosisDocumentGeneratorService;
+import uk.gov.hmcts.reform.fpl.service.docmosis.NoticeOfHearingGenerationService;
+import uk.gov.hmcts.reform.fpl.utils.ElementUtils;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static java.time.LocalDate.now;
+import static java.util.Comparator.comparing;
+import static org.apache.commons.lang3.ObjectUtils.defaultIfNull;
+import static uk.gov.hmcts.reform.fpl.enums.DocmosisTemplates.NOTICE_OF_HEARING;
+import static uk.gov.hmcts.reform.fpl.enums.YesNo.YES;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.DATE;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.asDynamicList;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ManageHearingsService {
+    private final NoticeOfHearingGenerationService noticeOfHearingGenerationService;
+    private final DocmosisDocumentGeneratorService docmosisDocumentGeneratorService;
+    private final UploadDocumentService uploadDocumentService;
+    private final HearingVenueLookUpService hearingVenueLookUpService;
+    private final HearingBookingService hearingBookingService;
+
+    public static final String FIRST_HEARING_FLAG = "firstHearingFlag";
+    public static final String HEARING_DATE_LIST = "hearingDateList";
+
+    public Map<String, Object> populateInitialFields(CaseData caseData) {
+        Map<String, Object> data = new HashMap<>();
+        List<Element<HearingBooking>> futureHearings = hearingBookingService.getFutureHearings(
+            defaultIfNull(caseData.getHearingDetails(), List.of()));
```

Comment:
again altering the getter in case data would make this more simple

Reply:
or just moving the whole thing to caseData like you suggested 😄